bot/roles.go

```diff
@@ -66,10 +70,32 @@ func roleLeaveCmd(ctx CommandCtx) error {
 	}
 	roleID := fmt.Sprintf("%d", roleIDInt)
 
+	if !checkRole(ctx, roleID) {
+		return nil
+	}
+
 	if err := ctx.Session.GuildMemberRoleRemove(msg.GuildID, msg.Author.ID, roleID); err != nil {
-		return fmt.Errorf("error removing role: %w", err)
+		return fmt.Errorf("could not remove role: %w", err)
 	}
 
 	log.Printf("Removed user with id: %s (%s) from %s role", msg.Author.ID, msg.Author.Username, roleID)
 	return ctx.Reply(fmt.Sprintf("You've been removed from the group %s.", args[0]))
 }
+
+func checkRole(ctx CommandCtx, givenRole string) bool {
+	mem, err := ctx.Session.State.Member(ctx.Message.GuildID, ctx.Message.Author.ID)
+	if err != nil {
+		if replyErr := ctx.Reply(fmt.Sprintf("could not find member: %s", err)); replyErr != nil {
+			log.Printf("failed to reply: %v", replyErr)
+		}
```

Comment:
Missing return in this block, right now this will print the log message and then crash when it runs the code below and `mem` is nil.

Reply:
This error check still doesn't contain a return (in the outside if) so the issue remains.